### PR TITLE
Don't clone fields unique to database

### LINKED ISSUE #3960: Permit Number Being Cloned
Permit Number should not be carried over since it's a unique field. Saving the permit leads to a business rule error.

https://github.com/specify/specify7/assets/53784701/bdbb2e9e-57f5-4625-9801-e1de82ed3705

[Specify 7 Crash Report - 2023-08-31T20_06_20.320Z.txt](https://github.com/specify/specify7/files/12489909/Specify.7.Crash.Report.-.2023-08-31T20_06_20.320Z.txt)

### PR DESCRIPTION
**Pull request by melton-jason**
Fixes #3960 

The following fields should not be carried over/cloned: 
> * Collectingevent uniqueIdentifier
> * CollectionObject uniqueIdentifier
> * Institution name
> * Locality uniqueIdentifier
> * Permit permitNumber
> * SpAppResourceData spAppResource
> * Specifyuser name

(Also, there was another issue in which global uniqueness rules were not working on the frontend. That fix should be included in #3962)

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/DataModel/resource.ts [hunk 1]
```diff
@@ -270,7 +270,8 @@ export const getUniqueFields = (model: SpecifyModel): RA<string> =>
     ...Object.entries(businessRuleDefs[model.name]?.uniqueIn ?? {})
       .filter(
         ([_fieldName, uniquenessRules]) =>
-          uniquenessRules in schema.domainLevelIds
+          uniquenessRules in schema.domainLevelIds ||
+          uniquenessRules === undefined
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/DataModel/resource.ts:274]: (I seen https://github.com/specify/specify7/issues/3960#issuecomment-1702819456)

can you double-check that this doesn't mark absolutely every field as unique, and rather only fields that have uniqueness constraints of some sort?

also, if you have a moment to spare, can you update/add the test cases?

**Review by CarolineDenis — APPROVED** [review]

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
When cloning accessions, I still get this error when there is an existing agent:
https://github.com/specify/specify7/issues/3967

Is this in the scope of this PR?

**Comment by melton-jason** [comment]
> When cloning accessions, I still get this error when there is an existing agent: #3967
> 
> Is this in the scope of this PR?

That issue (#3967) has been around since at _least_ `v7.8.10` (and possibly earlier, I just stopped checking previous versions).
I do not think there is enough overlap in underlying cause to justify that solution being included in this PR. 

_However_, it still could be included in the `v7.9` release in it's own PR if it is important enough (at the cost of further delaying the release)

**Review by grantfitzsimmons — APPROVED** [review]

**Comment by grantfitzsimmons** [comment]
> I do not think there is enough overlap in underlying cause to justify that solution being included in this PR.

I agree. Let's fix at a later time.

**Review by realVinayak — APPROVED** [review]

**Pull request merged**